src/Subscription.js

```diff
@@ -2,28 +2,34 @@ import EventEmitter from 'eventemitter3'
 import debugFactory from 'debug'
 import uniqueId from 'lodash.uniqueid'
 
-const debug = debugFactory('StreamrClient::Subscription')
-
 const DEFAULT_PROPAGATION_TIMEOUT = 5000
 const DEFAULT_RESEND_TIMEOUT = 5000
+
 /*
 'interface' containing the default parameters and functionalities common to every subscription (Combined, RealTime and Historical)
  */
 export default class Subscription extends EventEmitter {
     constructor(streamId, streamPartition, callback, groupKeys,
-        propagationTimeout = DEFAULT_PROPAGATION_TIMEOUT, resendTimeout = DEFAULT_RESEND_TIMEOUT) {
+        propagationTimeout = DEFAULT_PROPAGATION_TIMEOUT, resendTimeout = DEFAULT_RESEND_TIMEOUT, debug) {
         super()
-        if (!streamId) {
-            throw new Error('No stream id given!')
-        }
 
         if (!callback) {
             throw new Error('No callback given!')
         }
         this.streamId = streamId
         this.streamPartition = streamPartition
         this.callback = callback
-        this.id = uniqueId('sub')
+        const id = uniqueId('sub')
+        this.id = id
+        if (debug) {
+            this.debug = debug.extend(this.constructor.name).extend(id)
```

Comment:
Will `this.constructor.name` get changed to something intelligible by minification? I've encountered that sometimes when trying to use `this.constructor.name`, not sure if it's a problem with current tooling.

Reply:
Yeah it should be left as-is but I'll double-check

Reply:
Seems to keep the class name intact.